src/utils/getHydratedTemplateString.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { getHydratedTemplateString } from './getHydratedTemplateString';
+
+const getStringTemplate = (...args: any) => {
+  return args[0];
+};
+
+const removeLineBreaks = (styles: string[]) => {
+  return styles.map((style: string) =>
+    style.split('\n').join('').replace(/ /g, '')
+  );
+};
+
+const hasDifferentStyles = (
+  generatedStyles: string[],
+  expectedStyles: string[]
+) => {
+  const parsedGeneratedStyles = removeLineBreaks(generatedStyles);
+  const parsedExpectedStyles = removeLineBreaks(expectedStyles);
+
+  const hasDifferentStyle = parsedGeneratedStyles.filter(
+    (style: string, index: number) =>
+      style.trim() !== parsedExpectedStyles[index].trim()
+  );
+
+  return Boolean(hasDifferentStyle.length);
+};
+
+/**
+ * Test the core logic of converting Tailwind classes to their
+ * respective style equivalent.
+ *
+ * Testing done for the first argument of taged template literal
+ * which require transformation and is than sent over to styled tag
+ * from styled-components.
+ */
+describe('Tailwind Classes Hydration', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should generate correct styles for tailwind classes', () => {
+    const generatedStyles = getHydratedTemplateString`
+        .p-2;
+        .mt-2;
+      `;
+
+    const expectedStyles = getStringTemplate`
+        padding: 0.5rem;
+        margin-top: 0.5rem;
+      `;
+    expect(hasDifferentStyles(generatedStyles, expectedStyles)).toBeFalsy();
```

Comment:
Loved this testing style ❤️

Reply:
Thanks :) 
Wasnt super sure if this was the correct way to test but glad that you liked it